integreat_cms/cms/forms/pages/page_form.py

```diff
@@ -102,6 +103,11 @@ def __init__(self, **kwargs):
         # Let mirrored page queryset be empty per default and only fill it if a region is selected
         mirrored_page_queryset = Page.objects.none()
 
+        # Filter the offer provides available for embedding
+        self.fields["page_based_offer"].queryset = self.instance.region.offers.filter(
+            supported_by_app_in_content=True
+        )
```

Comment:
Since we only allow offers here that are available in the current region: Do we also need a post-save signal handler for the region model to remove an offer from all pages of that region when it's disabled in the region form?

Reply:
That's a good question I had not thought about yet. Actually I am not sure removing the offers is ideal - for example, what in case of accidental disables of the offer? That could potentially caue some frustration. I would throw
- just keep them embedded (plus maybe show a warning in the `OfferTemplate` form) and
- disable the disabling unless no page embeds those offers anymore
into the ring of potential solutions.

Reply:
Hmm yeah I agree this isn't ideal. But if the offer is really disabled, it could also be convenient to do this via the region form without having to click though all pages and remove it manually everywhere?
Hmm, not sure about this. I guess as quick fix it's safest to disable the disabling (:grin:) until we come up with a long term solution.

Reply:
Alright, sounds good to me!